### PR TITLE
#12339 Add documentation about sponsors

### LINKED ISSUE #12339: Document sponsorship process
We had some old Trac wiki page that is now archived at https://github.com/twisted/trac-wiki-archive/blob/trunk/TwistedSoftwareFoundation.mediawiki

Try to migrate it to the Twisted Documentation.

------

We have https://twisted.org/ which is our "presentational" website

But I think that the current Read The Docs, documentation site, is the main site for twisted/twisted

----

And this is about documenting the sponsorship process .. so I think that it makes sense to have this information in the doc site and not on the presentational site

### PR DESCRIPTION
**Pull request by adiroiban**
## Scope and purpose

Fixes #12339

This is based on the old Trac wiki page https://github.com/twisted/trac-wiki-archive/blob/trunk/TwistedSoftwareFoundation.mediawiki

and Glyph email https://mail.python.org/archives/list/twisted@python.org/thread/55VR6KWJ46WE4H3B7QOAWE53UT4LRQBW/

## How to test

* Check README rendering and content here https://github.com/twisted/twisted/tree/12339-sponsors-documentation 
* Check documentation sponsors https://twisted--12340.org.readthedocs.build/en/12340/development/sponsorship.html
* Check documentation start page https://twisted--12340.org.readthedocs.build/en/12340/
* Check documentation dev index page https://twisted--12340.org.readthedocs.build/en/12340/development/index.html

### CODE REVIEW

File: docs/development/coding-standard.rst [hunk 1]
```diff
@@ -1,5 +1,5 @@
-Twisted Coding Standard
-=======================
+Coding Standard
```
Comment (adiroiban) [inline docs/development/coding-standard.rst:1]: This was just a drive-by comment...as in the table of content each item was starting with "Twisted ..." and it was kind of redundeand

File: docs/index.rst [hunk 2]
```diff
@@ -27,3 +25,17 @@ Contents:
    api/index
    GitHub <https://github.com/twisted/twisted>
    PyPI <https://pypi.org/project/twisted>
+
+
+Sponsors
```
Comment (adiroiban) [inline docs/index.rst:30]: I have also added the sponsors on the documentation start page.

I guess that it's ok... it should not be too much spam

It's at the bottom, so the focus on the documentation site is still to provide documentation for end users


but I think that we should provide enough benefits for companies to consider sponsoring Twisted

**Comment by adiroiban** [comment]
this is ready for review

@glyph I hope that it's ok that I have used the text from your email to the mailing list as the content of the new sponsors page

needs-review

File: README.rst [hunk 3]
```diff
@@ -121,3 +132,9 @@ Again, see the included `LICENSE <LICENSE>`_ file for specific legal details.
 
 .. |rtd| image:: https://readthedocs.org/projects/twisted/badge/?version=latest&style=flat
 .. _rtd: https://docs.twistedmatrix.com
+
+.. |thinkst| image:: https://github.com/user-attachments/assets/a5b52432-2d18-4d91-a3c9-772fb2e02781
+.. _thinkst: https://thinkst.com/
+
+.. |sftpplus| image:: https://github.com/user-attachments/assets/5f585316-c7e8-4ef1-8fbb-923f0756ceed
+.. _sftpplus: https://www.sftpplus.com/
```
Comment (twm) [inline README.rst]: These should have alt text:

```suggestion
.. |thinkst| image:: https://github.com/user-attachments/assets/a5b52432-2d18-4d91-a3c9-772fb2e02781
    :alt: Thinkst Canary
    :target: https://thinkst.com/

.. |sftpplus| image:: https://github.com/user-attachments/assets/5f585316-c7e8-4ef1-8fbb-923f0756ceed
    :alt: SFTPPlus
    :target: https://www.sftpplus.com/
```

The same applies to the other images, but not a blocker.

File: docs/development/sponsorship.rst [hunk 4]
```diff
@@ -0,0 +1,185 @@
+Sponsor the project
```
Comment (twm) [inline docs/development/sponsorship.rst]: The other page titles have Title Case:

```suggestion
Sponsor the Project
```

File: docs/development/sponsorship.rst [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+   We have empty lines between sponsors of different levels,
+   to render them as separate rows.
+
+|sftpplus|_
+
+
+How to donate
+-------------
+
+Since April 2023, the Twisted project got fiscal sponsorship from the `Python Software Foundation <https://www.python.org/psf-landing/>`_.
```
Comment (twm) [inline docs/development/sponsorship.rst]:
```suggestion
Since April 2023, the Twisted project has been fiscally sponsored by the `Python Software Foundation <https://www.python.org/psf-landing/>`_.
```

File: docs/development/sponsorship.rst [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+* `GitHub Sponsors <https://github.com/sponsors/twisted/>`_
+* `Tidelift <https://tidelift.com/lifter/search/pypi/Twisted>`_
+
+Donations made directly via the Python Software Foundation (PSF) are tax-deductible in USA.
+The PSF is a recognized 501(c)(3) non-profit organization.
+
+GitHub Sponsors allows monthly donations or one time donations.
+
+For the contributions made using Tidelift we don't have any insights.
+At the moment, we receive about $300 per month via Tidelift.
```
Comment (twm) [inline docs/development/sponsorship.rst]: Perhaps should date this?

```suggestion
As of October 2024, we receive about $300 per month via Tidelift.
```

File: docs/development/sponsorship.rst [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+* in the release notes
+
+
+$200 per month or equivalent
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+With $200 per month or the equivalent $2400/year we can have your logo:
+
+* In the README file of the twisted/twisted project.
+* On the twisted.org website in the Donation section
```
Comment (twm) [inline docs/development/sponsorship.rst]: The section is actually titled "Twisted Sponsors":

image

File: docs/development/sponsorship.rst [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+
+* in the release notes
+* other communications
+* merchandise
+
+
+Logos
+-----
+
+Sponsors logos are displayed on the following sites:
```
Comment (twm) [inline docs/development/sponsorship.rst]:
```suggestion
Sponsor logos are displayed on the following sites:
```

File: docs/development/sponsorship.rst [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+Logos can link to any URL suggested by the sponsors.
+
+
+Non-monetary donations and contributions
+----------------------------------------
+
+The Twisted Project will consider the value of non-monetary donations to the project, for example, donations of hardware, software licenses, or hosting - on a case-by-case basis.
+
+If your company is directly contributing code development or code review,
+we encourage to use company email addresses.
```
Comment (twm) [inline docs/development/sponsorship.rst]:
```suggestion
If your company is directly contributing code development or code review,
we encourage you to use a company email address.
```

File: docs/development/sponsorship.rst [hunk 10]
```diff
@@ -0,0 +1,185 @@
...
+   You can then take a screenshot / capture of the logo in PNG format.
+   You can upload the PNG logos via GitHub Issues, for example as part of the
+   GitHub Issue that was created to add a new sponsor.
+   Once the file is uploaded, you can get the link.
+
+.. |thinkst| image:: https://github.com/user-attachments/assets/a5b52432-2d18-4d91-a3c9-772fb2e02781
+.. _thinkst: https://thinkst.com/
+
+.. |sftpplus| image:: https://github.com/user-attachments/assets/5f585316-c7e8-4ef1-8fbb-923f0756ceed
+.. _sftpplus: https://www.sftpplus.com/
```
Comment (twm) [inline docs/development/sponsorship.rst]: Same alt text comment applies here.

File: docs/index.rst [hunk 11]
```diff
@@ -27,3 +25,17 @@ Contents:
...
+
+|thinkst|_
+
+|sftpplus|_
+
+.. |thinkst| image:: https://github.com/user-attachments/assets/a5b52432-2d18-4d91-a3c9-772fb2e02781
+.. _thinkst: https://thinkst.com/
+
+.. |sftpplus| image:: https://github.com/user-attachments/assets/5f585316-c7e8-4ef1-8fbb-923f0756ceed
+.. _sftpplus: https://www.sftpplus.com/
```
Comment (twm) [inline docs/index.rst:43]: And here as well.

**Review by twm — APPROVED** [review]
This looks good to me, though I noted some minor grammatical issues and I strongly suggest adding alt text to the logos!

The "upload an image in a GitHub issue, then hotlink it" technique seems like it might break if GitHub makes changes on their end, but I think that we can deal with that as it happens.

File: docs/development/sponsorship.rst (hunk 7 above)
Comment (adiroiban) [reply docs/development/sponsorship.rst]: True
```suggestion
* On the twisted.org website in the "Sponsors" section
```

File: docs/development/sponsorship.rst (hunk 10 above)
Comment (adiroiban) [inline docs/development/sponsorship.rst]:
```suggestion
.. |thinkst| image:: https://github.com/user-attachments/assets/a5b52432-2d18-4d91-a3c9-772fb2e02781
    :alt: Thinkst Canary
    :target: https://thinkst.com/

.. |sftpplus| image:: https://github.com/user-attachments/assets/5f585316-c7e8-4ef1-8fbb-923f0756ceed
    :alt: SFTPPlus
    :target: https://www.sftpplus.com/
```

File: docs/index.rst (hunk 11 above)
Comment (adiroiban) [inline docs/index.rst]:
```suggestion
.. |thinkst| image:: https://github.com/user-attachments/assets/a5b52432-2d18-4d91-a3c9-772fb2e02781
    :alt: Thinkst Canary
    :target: https://thinkst.com/

.. |sftpplus| image:: https://github.com/user-attachments/assets/5f585316-c7e8-4ef1-8fbb-923f0756ceed
    :alt: SFTPPlus
    :target: https://www.sftpplus.com/
```

**Comment by adiroiban** [comment]
Many thanks Tom for the review

> This looks good to me, though I noted some minor grammatical issues and I strongly suggest adding alt text to the logos!

I have applied your changes.


> The "upload an image in a GitHub issue, then hotlink it" technique seems like it might break if GitHub makes changes on their end, but I think that we can deal with that as it happens.

We can copy the logos is `docs/_static` ... but I was not sure how well the work with the README rendering.

I will try to move the files there once we got a new sponsor ... or if GitHub Issues links are failing
